transcendence-app/src/user/user.service.ts

```diff
@@ -1,21 +1,22 @@
 import { Injectable } from '@nestjs/common';
 import { UserDto } from './dto/user.dto';
 import { UserEntity } from './user.entity';
+import { v4 as uuidv4 } from 'uuid';
 
 @Injectable()
 export class UserService {
   private readonly users: UserEntity[] = [];
 
   private findOne(username: string) {
-    return this.users.find((user) => user.Dto.username === username);
+    return this.users.find((user) => user.username === username) || null;
```

Comment:
Why is it better to return null than undefined?

Reply:
`null` is an intentional lack of content (controlled) `undefined` is a non-intentional lack of content (not-controlled)

Reply:
Obviously I understand the concept as in C, and it would definitely be my approach in C (because there is not undefined as a value/type) but I'm not sure it's the best solution here, (except for the explanation from gbudau about deserializeUser).

Alas, the method we are wrapping (.find() ) returns an undefined, because there is no defined member of this array that corresponds to the key we are trying to match. We are changing the semantics when we decide to return a null instead.